drivers/drivers.go

```diff
@@ -0,0 +1,107 @@
+// Package drivers abstracts different object storages, such as local, s3, ipfs
+package drivers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/net"
+)
+
+// Storages configured at start (used by broadcaster)
+var Storages []OSDriver
+
+// ExternalStorage own external storage, if configured
+var ExternalStorage OSDriver
+
+// LocalStorage single instance of local storage
+var LocalStorage *localOS
+
+// OSDriver common interface for Object Storage
+type OSDriver interface {
+
+	// StartSession session is unique identified by nonce, should return existing session
+	StartSession(jobID int64, manifestID string, nonce uint64) OSSession
+
+	// IsExternal
+	IsExternal() bool
+}
+
+// OSSession ...
+type OSSession interface {
+	SaveData(streamID, name string, data []byte) (*net.TypedURI, string, error)
```

Comment:
Seems like TypedURI includes the abs that's the second return value.  Should we change this signature to
```suggestion
	SaveData(streamID, name string, data []byte) (uri *net.TypedURI, err error)
```

Reply:
I'm not even sure that TypedURI is needed; we'd probably be just fine returning a string or regular golang net/URL.

Reply:
@j0sh I've added TypeURI so we do not parse data from string url. Parsing urls implies that url has some specific structure, but with that we don't have benefits of static type checking.